docs/spec/staking/spec-technical.md

```diff
@@ -154,76 +169,82 @@ Description:
  - AdjustmentFeePool: Adjustment factor used to passively calculate each bonds
    entitled fees from `GlobalState.FeePool`
  - AdjustmentRewardPool: Adjustment factor used to passively calculate each
-   bonds entitled fees from `Candidate.ProposerRewardPool``
+   bonds entitled fees from `Candidate.ProposerRewardPool`
+      
  
 ### QueueElem
 
 Unbonding and re-delegation process is implemented using the ordered queue data structure. 
-All queue elements used share a common structure:
+All queue elements share a common structure:
 
 ``` golang
 type QueueElem struct {
 	Candidate   crypto.PubKey
-	InitHeight  int64    // when the queue was initiated
+	InitTime    int64    // when the element was added to the queue
 }
 ```
 
-The queue is ordered so the next to unbond/re-delegate is at the head. Every
+The queue is ordered so the next element to unbond/re-delegate is at the head. Every
 tick the head of the queue is checked and if the unbonding period has passed
-since `InitHeight`, the final settlement of the unbonding is started or re-delegation is executed, and the element is
+since `InitTime`, the final settlement of the unbonding is started or re-delegation is executed, and the element is
 pop from the queue. Each `QueueElem` is persisted in the store until it is popped from the queue. 
 
 ### QueueElemUnbondDelegation
 
+QueueElemUnbondDelegation structure is used in the unbonding queue. 
+
 ``` golang
 type QueueElemUnbondDelegation struct {
 	QueueElem
-	Payout           Address  // account to pay out to
-    Shares           rational.Rat  // amount of delegator shares which are unbonding
-    StartSlashRatio  rational.Rat  // candidate slash ratio at start of re-delegation
+	Payout           Address       // account to pay out to
+    Tokens           coin.Coins    // the value in Atoms of the amount of delegator shares which are unbonding
+    StartSlashRatio  rational.Rat  // candidate slash ratio 
 }
 ``` 
-In the unbonding queue - the fraction of all historical slashings on
-that validator are recorded (`StartSlashRatio`). When this queue reaches maturity
-if that total slashing applied is greater on the validator then the
-difference (amount that should have been slashed from the first validator) is
-assigned to the amount being paid out.  
+
+TODO: Explain what is StartSlashRation.
 
 ### QueueElemReDelegate
 
+QueueElemReDelegate structure is used in the re-delegation queue. 
+
 ``` golang
 type QueueElemReDelegate struct {
 	QueueElem
-	Payout       Address  // account to pay out to
+	Payout       Address       // account to pay out to
     Shares       rational.Rat  // amount of shares which are unbonding
     NewCandidate crypto.PubKey // validator to bond to after unbond
 }
 ```
+Q: Why we need Payout address in the re-delegation element?
```

Comment:
I think the payout address is the address to re-delegate to... Maybe we should rename to something like `Receiving`